src/field_arithmetic.rs

```diff
@@ -35,10 +35,10 @@ impl<F: Field> GadgetBuilder<F> {
     }
 
     /// `x^p` for a constant `p`.
-    pub fn exp(&mut self, x: &Expression<F>, p: &Element<F>) -> Expression<F> {
+    pub fn exponentiation(&mut self, x: &Expression<F>, p: &Element<F>) -> Expression<F> {
```

Comment:
Renamed to `exponentiation` for notation consistency away from references to `Expression`.